### PR TITLE
Chore: Migrate ActionSheets to `react-native-bottom-sheet` and make them dynamic

### PR DESCRIPTION
**Pull request by try-catch-stack**
## Proposed changes

* Migrated ActionSheets to [react-native-bottom-sheet](https://gorhom.github.io/react-native-bottom-sheet/) .
* Updated the logic to render other components like modals in the Action Sheet.

#### Pay attention on these tests for landscape and tablets (action sheet should be center aligned there)
- Hold message to show action sheet and make sure it's working
- Go to Members and tap on a member. The action sheet should be working.
- Tap `+` on Messagebox to open the action sheet and make sure it's working.
- Example of calling ActionSheet with some custom component :
```
const styles = StyleSheet.create({
			heading: {
				color: 'white',
				fontSize: 20,
				textAlign: 'center'
			},
			buttonsContainer: {
				flexDirection: 'row',
				justifyContent: 'center'
			},
			button: {
				width: 150,
				marginHorizontal: 20
			},
			input: {
				marginVertical: 20,
				color: 'gray',
				fontSize: 16,
				lineHeight: 20,
				padding: 8,
				marginHorizontal: 35,
				backgroundColor: colors.bodyText
			}
		});
		showActionSheet({
			children: (
				<View>
					<Text style={styles.heading}>Some custom component</Text>
					<BottomSheetTextInput style={styles.input} placeholder='Enter some text' />
					<View style={styles.buttonsContainer}>
						<Button
							title='Cancel'
							type='secondary'
							style={styles.button}
							backgroundColor={colors.dangerColor}
							theme={theme}
						/>
						<Button
							title='Save'
							type='primary'
							style={styles.button}
							backgroundColor={colors.successColor}
							theme={theme}
						/>
					</View>
				</View>
			),
			snaps: ['30%']
		});
```
image

## Screenshots

https://user-images.githubusercontent.com/77742477/168103080-c82017c8-9029-4c44-96c8-ee49fcdf02e9.mp4

Rendering custom component inside ActionSheet :

https://user-images.githubusercontent.com/77742477/168123492-9492fb0a-a5b3-4322-9570-f24441106c82.mp4

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [X] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [X] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [X] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/containers/ActionSheet/ActionSheet.tsx [hunk 1]
```diff
@@ -122,69 +102,38 @@ const ActionSheet = React.memo(
...
-							style={{ backgroundColor: themes[theme].focusedBackground }}
-							contentContainerStyle={styles.content}
-							ItemSeparatorComponent={List.Separator}
-							ListHeaderComponent={List.Separator}
-							ListFooterComponent={renderFooter}
-							getItemLayout={getItemLayout}
-							removeClippedSubviews={isIOS}
-						/>
-					</>
+					<BottomSheet
```
Comment (diegolmello) [inline app/containers/ActionSheet/ActionSheet.tsx:124]: image
It should be centralized. You can test it on tablets as well.

File: app/containers/ActionSheet/ActionSheet.tsx [hunk 2]
```diff
@@ -39,27 +28,26 @@ const ANIMATION_CONFIG = {
...
 		const [isVisible, setVisible] = useState(false);
 		const { height } = useDimensions();
 		const { isLandscape } = useOrientation();
 		const insets = useSafeAreaInsets();
 
-		const maxSnap = Math.max(
-			height -
-				// Items height
-				ITEM_HEIGHT * (data?.options?.length || 0) -
+		const maxSnap = Math.min(
```
Comment (diegolmello) [inline app/containers/ActionSheet/ActionSheet.tsx:39]: We could simplify this logic by using [useBottomSheetDynamicSnapPoints])(https://gorhom.github.io/react-native-bottom-sheet/hooks#usebottomsheetdynamicsnappoints).
This code here is working (your proposal too), but it's using lots of static info.
Maybe we could use that hook, remove this complex logic from our component and only say our snaps are `['50%', 'CONTENT_HEIGHT']`

File: app/containers/ActionSheet/BottomSheetContent.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+				accessibilityLabel={I18n.t('Cancel')}>
+				<Text style={[styles.text, { color: colors.bodyText }]}>{I18n.t('Cancel')}</Text>
+			</Button>
+		) : null;
+
+	const renderItem = ({ item }: { item: IActionSheetItem['item'] }) => <Item item={item} hide={hide} />;
+
+	if (options) {
+		return (
+			<BottomSheetFlatList
```
Comment (diegolmello) [inline app/containers/ActionSheet/BottomSheetContent.tsx:39]: The expected UX is not to scroll on the FlatList, but to move the whole action sheet.
On this video, you can see it's scrolling if I touch inside and I need to scroll from the handle to actually scroll the action sheet.

https://user-images.githubusercontent.com/804994/168338394-0b3ae8b2-56e0-4bbe-a936-54074d23c0ca.mp4

**Review by diegolmello — COMMENTED** [review]
I've left a few comments here.
Also I couldn't test the use case without options.
Maybe you could post a reproducible example on the PR description?
It could be very simple just to test it.

You're almost there and it's very good already.
Nice job!! 🚀

File: app/containers/ActionSheet/ActionSheet.tsx (hunk 1 above)
Comment (try-catch-stack) [reply app/containers/ActionSheet/ActionSheet.tsx:124]: Done

Screenshot_1652526699

**Review by diegolmello — APPROVED** [review]
LGTM! 👏🏼 👏🏼 👏🏼

**Pull request merged**